src/test/kotlin/io/github/nomisrev/routes/TagRouteSpec.kt

```diff
@@ -0,0 +1,24 @@
+package io.github.nomisrev.routes
+
+import io.github.nomisrev.withServer
+import io.kotest.core.spec.style.StringSpec
+import io.kotest.matchers.collections.shouldHaveSize
+import io.kotest.matchers.shouldBe
+import io.ktor.client.call.body
+import io.ktor.client.request.get
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.contentType
+
+
+class TagRouteSpec :
+  StringSpec({
+    "can get all tags" {
+      withServer {
+        val response = get("/tags") { contentType(ContentType.Application.Json) }
+
+        response.status shouldBe HttpStatusCode.OK
+        response.body<TagResponse>().tags shouldHaveSize 0
+      }
```

Comment:
To be honest, I'm a bit lost here. I'm not sure how to implement this test. If you don't mind, could you share an example?

Reply:
This is probably the file with the most examples: https://github.com/nomisRev/ktor-arrow-example/blob/main/src/test/kotlin/io/github/nomisrev/routes/UserRouteSpec.kt

but let me explain what is going on:
1. There is a _project wide_ test container running when the test-suite starts, and ends which runs the Postgres database.
2. There is a project-wide instance of the `Dependencies` based on the testcontainers setup, and both 1 and 2 happen inside [`KotestProject`](https://github.com/nomisRev/ktor-arrow-example/blob/main/src/test/kotlin/io/github/nomisrev/KotestProject.kt)
3. [`withServer`](https://github.com/nomisRev/ktor-arrow-example/blob/962b1b549c63d12b15dd444faec3d336d0468577/src/test/kotlin/io/github/nomisrev/ktor.kt#L15) hides all the `ktor-test` boilerplate to start the test server, and initialises the `app` (all routes, and server configuration).
4. This is inside your test, you have access to the configured `HttpTestClient` and just write your test as usual as you've done here.

There is two things missing in this test suite:
 1. We should test the behavior if `tags` are available. Tags can be inserted through the `ArticleService`, which you have access to from `withServer { dependencies -> }`. So first we should insert some tags, by creating an article with tags, and then check if they're available for retrieval.
 2. But the setup is not yet ready for this, we should be able to modify the `dependencies` such that we can _fake_ a unexpected error.

I think `2` warrants a new issue, because those tests are also missing from `UserRouteSpec` and requires modifying more code so I think it's out of the scope of this ticket. If you're interested in this, or picking up another ticket I'd be more than happy to help ☺️